File: my_py_pkg/my_py_pkg/smartphone.py
import rclpy 
from rclpy.node import Node 
from example_interfaces.msg import String

class Smartphone(Node):
    def __init__(self): 
        super().__init__("smartphone") 
        self.get_logger().info("Smartphone has been started")
        self.subscriber = self.create_subscription(
            String, 
            "robot_news", # publisher topic name 
            self.callback_robot_news,
            10
        )

    def callback_robot_news(self, msg): 
        self.get_logger().info(msg.data)
    
def main(args=None): 
    rclpy.init(args=args)
    node = Smartphone() 
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == "__main__": 
    main()

File: my_cpp_pkg/src/smartphone.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"
#include <string> 

class Smartphone: public rclcpp::Node 
{ 
    public: 
        Smartphone(): Node("smartphone")
        { 
            RCLCPP_INFO(this->get_logger(), "Hello this is subscriber");  
        }

    private: 

        rclcpp::Subscription<example_interfaces::msg::String>::SharedPtr subscriber;

}; 


int main(int argc, char** argv)
{ 
    rclcpp::init(argc, argv); 
    auto node = std::make_shared<Smartphone>(); 
    rclcpp::spin(node); 
    rclcpp::shutdown();
    return 0;
};